core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -782,16 +782,22 @@ RexNode simplifyAnd2(List<RexNode> terms, List<RexNode> notTerms) {
       }
       notSatisfiableNullables.add(notDisjunction);
     }
+
     if (notSatisfiableNullables != null) {
+      // Remove the intersection of "terms" and "notTerms"
       terms.removeAll(notSatisfiableNullables);
-      terms.add(rexBuilder.makeNullLiteral(notSatisfiableNullables.get(0).getType()));
+      notTerms.removeAll(notSatisfiableNullables);
+
+      // The intersection simplify to "null and x1 is null and x2 is null..."
+      List<RexNode> nullAndIsNullTerms = new ArrayList<>();
```

Comment:
Please remove this list.
existing `terms` is just fine.

Reply:
I add this list is just to make the code more clearly in the first.Technically,this list is duplication.I will remove it.